### PR TITLE
Added Copy Source Tags

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/autorest.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/564d0137d08defea63d9de2413ef5336fbfe2e6d/specification/storage/data-plane/Microsoft.BlobStorage/preview/2021-04-10/blob.json
+    - C:\Users\seanmcc\git\azure-rest-api-specs\specification\storage\data-plane\Microsoft.BlobStorage\preview\2021-04-10\blob.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/src/autorest.md:7]: TODO undo this.

File: sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs [hunk 2]
```diff
@@ -440,6 +441,11 @@ public partial class BlobCopyInfo
         public System.DateTimeOffset LastModified { get { throw null; } }
         public string VersionId { get { throw null; } }
     }
+    public enum BlobCopySourceTags
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs:444]: Consider changing to BlobCopySourceTagsMode or BlobCopyTagsMode.

File: sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs [hunk 3]
```diff
@@ -1089,6 +1095,7 @@ public partial class BlobSyncUploadFromUriOptions
         public Azure.Storage.Blobs.Models.AccessTier? AccessTier { get { throw null; } set { } }
         public byte[] ContentHash { get { throw null; } set { } }
         public bool? CopySourceBlobProperties { get { throw null; } set { } }
+        public Azure.Storage.Blobs.Models.BlobCopySourceTags? CopySourceTags { get { throw null; } set { } }
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs:1098]: Consider changing to BlobCopySourceTagsMode or BlobCopyTagsMode.

File: sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs [hunk 4]
```diff
@@ -420,6 +420,7 @@ public partial class BlobCopyFromUriOptions
     {
         public BlobCopyFromUriOptions() { }
         public Azure.Storage.Blobs.Models.AccessTier? AccessTier { get { throw null; } set { } }
+        public Azure.Storage.Blobs.Models.BlobCopySourceTags? CopySourceTags { get { throw null; } set { } }
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs:423]: Consider changing to BlobCopySourceTagsMode or BlobCopyTagsMode.

File: sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs:444]: Fixed.

File: sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs (hunk 4 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs:423]: fixed.

File: sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs:1100]: fixed.

**Review by amnguye — APPROVED** [review]

**Pull request merged**